### DISCUSSION TITLE
How to customize Admin UI (Dashboard)

### DISCUSSION THREAD

**Discussion by megaacheyounes** [vendurehq/vendure#2784] [Q&A]
Hello

Thank you for this amazing project.

I've been trying Vendure and so far it seemed like a great platform, however, I've been trying to customize the Admin UI (Dashboard) as I do not like theme colors and some layouts/components. I wanted to know the best approach to make changes to the UI.

for example, for the login page, I tried editing the complied code in `node_modules/@vendure/admin-ui` and use the command `patch-package` to apply the changes during deployment, this did work but it seemed inefficient and not ideal.

before:
 
<img width="500" alt="before" src="https://github.com/vendure-ecommerce/vendure/assets/36759976/fac90dbf-c513-4811-94ae-14a2e127ac04">

after :
<img width="500" alt="vendure_after" src="https://github.com/vendure-ecommerce/vendure/assets/36759976/5a64a86e-cab7-4c09-a5e9-624238e0c7bd">

May I know the best approach to modify the Admin UI ? can I move the `admin-ui` out of `node_modules`, modify the source and include it as a custom plugin? or use `patch-package` as described above? or is there a better approach?

Thank you!

**Comment by michaelbromley** [comment]
Hi,

The supported ways to customize the Admin UI are covered in this section of the docs: https://docs.vendure.io/guides/extending-the-admin-ui/admin-ui-theming-branding/

**Reply by megaacheyounes** [reply]
Thanks, but what about advanced customizations, like modifying HTML structure ?

**Reply by michaelbromley** [reply]
In that entire docs section on "Extending the Admin UI" it gives all the ways the UI can be modified. 

For swapping out the login screen, it _might_ be possible with this feature: https://docs.vendure.io/guides/extending-the-admin-ui/defining-routes/#overriding-built-in-routes

I've not tested that use-case though.
